design/ep-1248.md

```diff
@@ -0,0 +1,252 @@
+
+## Goal
+
+Based on the capabilities of the already implemented KV Cache Sidecar, this proposal aims to build and integrate an **optional** context caching feature for the Aibrix system. This feature will allow users to efficiently reuse the model inference's KV Cache via a session ID, thereby significantly reducing redundant computation overhead and optimizing overall performance and resource consumption in multi-turn or conversational LLM interactions.
+
+## Implementation
+
+### Request Flow
+
+We will introduce a new endpoint: `/v1/context` to manage context caches. The following fields will be used:
+
+  - `session_id`: A unique identifier for each context cache, created upon the first request and used in subsequent requests.
+  - `ttl`: The time-to-live for the cache, after which it will be automatically cleared.
+
+#### Creating a Cache for a Session
+
+```mermaid
+sequenceDiagram
+    participant C as Client
+    participant E as Envoy
+    participant G as Gateway Plugin
+    participant R as Context Cache Manager
+    participant IP as InferencePod
+    participant V as vLLM Main Container
+    participant S as KV Cache Sidecar
+
+    C->>+E: POST /v1/context (prompt, model, ttl)
+    E->>+G: Forward Request
+    G->>+R: 1. Request Session ID & Metadata Creation
+    R->>-G: Return Session ID
+    G->>+V: 2. Submit Prompt for Initial Inference
+    V->>V: 2.1. Compute KV Cache for Prompt
+    V-->>V: 2.2. Generate Completion (if needed)
+    V->>+S: 3. Export & Store KV Cache (via Sidecar API/IPC)
+    S->>S: 3.1. Persist KV Cache Data
+    S->>-V: Confirmation of Storage & Sidecar Cache ID
+    V->>-G: 4. Return Initial Inference Response (incl. Sidecar Cache ID)
+    G->>+R: 5. Register Session Metadata (session_id, Sidecar Cache ID, TTL)
+    R->>-G: Confirmation of Registration
+    G->>-E: Pipe back Response (with session_id, usage)
+    E->>-C: Complete Response
+    Note over R,S: Context Cache Manager manages metadata. Sidecar handles actual KV Cache data.
+```
+
+Before using context caching, users first need to create it. Here, we create a context cache with a `ttl` of one hour.
+
+```shell
+curl -X POST http://localhost:8000/v1/context \
+  -H "Content-Type: application/json" \
+  -H "Authorization: Bearer test-key-1234567890" \
+  -d '{
+     "model": "facebook-opt-125m",
+     "prompt": "Say this is a test",
+     "ttl": 3600,
+   }'
+```
+
+In the response, we can obtain the unique identifier for the created session, `session_id`.
+
+```json
+{
+  "id": "cmpl-de1f99972bd34149968489cb100b2c88",
+  "object": "text_completion",
+  "created": 1752594611,
+  "model": "facebook-opt-125m",
+  "session_id": "session-01"
```

Comment:
same problem here. do you expect the engine to return the session_id, it's not supported yet as I know?